pipeline/PRGate.groovy

```diff
@@ -0,0 +1,44 @@
+node{
+    timestamps{
+        deleteDir()
+        def pull_id = env.ghprbPullLink.tokenize('/')[-1];
+        currentBuild.description = "<a href=${env.ghprbPullLink}> PR #${pull_id}: ${ghprbPullTitle}</a>";
+
+        checkout(
+        [$class: 'GitSCM', branches: [[name: 'master']],
+        extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'on-build-config']],
+        userRemoteConfigs: [[url: 'https://github.com/RackHD/on-build-config']]])
+        String library_dir="$WORKSPACE/on-build-config"
+
+        shareMethod = load("on-build-config/jobs/ShareMethod.groovy")
+
+        try{
+            String repo_dir = ""
+            stage("Check Out PR"){
+                dir("image-service"){
+                    checkout scm
+                }
+                repo_dir = "$WORKSPACE/image-service"
+            }
+            
+            stage("Unit Test"){
+                unit_test = load("$repo_dir/pipeline/UnitTest.groovy")
+                unit_test.runTest(library_dir)
+            }
+            
+            stage("FunctionTest"){
+                function_test = load("$repo_dir/pipeline/FunctionTest.groovy")
+                function_test.runTest(library_dir)
+            }
+            currentBuild.result="SUCCESS"
+            
+        } finally{
+            stage("Write Back"){
+                String manifest_path = "$WORKSPACE/manifest.json"
+                shareMethod.generateManifestFromPR(manifest_path)
```

Comment:
I didn't find this function defined in https://github.com/RackHD/on-build-config/blob/master/jobs/ShareMethod.groovy